server/rooms.js

```diff
@@ -1693,8 +1693,12 @@ let Rooms = Object.assign(getRoom, {
 				user.inviteOnlyNextBattle = false;
 			}
 		}
-		if (options.tour && !room.tour.modjoin) inviteOnly = [];
-		if (inviteOnly.length) {
+		const prefix = /** @type {RoomBattle} */(room.game).forcedPublic();
+		if (prefix && inviteOnly.length) {
+			room.isPrivate = false;
+			room.modjoin = null;
+			room.add(`|raw|<div class="broadcast-red"><strong>This battle is required to be public due to a player having a name prefixed by '${prefix}'.</div>`);
```

Comment:
Apparently we don't provide a warning for this in `room.tour` situations, but I really think player's should be alerted if their `/ionext` doesn't work all of a sudden.

Reply:
Why does this need to be a broadcast instead of an error reply (like in `/modjoin`)?

Reply:
This is the is the same as what `/ionext` does below. It doesnt happen when you issue the command (which is when we `errorReply`, it happens after the command already succeeded, which is why I think the more aggressive messaging is important.